tests/connectors/api_recaptcha_test.py

```diff
@@ -1,127 +1,26 @@
-    }
-)
-bad_request_return_value = MagicMock(status_code=400, text="")
-invalid_recaptcha_without_specific_error_return_value = MagicMock(status_code=200, text="")
-invalid_recaptcha_without_specific_error_return_value.json = MagicMock(
-    return_value={
-        "success": False,
-    }
-)
-wrong_action_return_value = MagicMock(status_code=200, text="")
-wrong_action_return_value.json = MagicMock(
-    return_value={
-        "success": True,
-        "score": 0.9,
-        "action": "send",
-        "challenge_ts": datetime.datetime.now().strftime("%Y-%m-%dT%H:%M:%S%Z"),
-        "hostname": "url",
-    }
-)
-errors_return_value = MagicMock(status_code=200, text="")
-errors_return_value.json = MagicMock(
-    return_value={
-        "success": False,
-        "error-codes": ["timeout-or-duplicate", "unknown-error"],
-    }
-)
-test_api_recaptcha_exceptions_data = {
-    "bad_request": (bad_request_return_value, "Couldn't reach recaptcha api: 400 "),
-    "invalid_request": (invalid_recaptcha_without_specific_error_return_value, "This is not a valid reCAPTCHA token"),
-    "wrong_action": (wrong_action_return_value, "The action 'send' does not match 'submit' from the form"),
-    "errors": (
-        errors_return_value,
-        "Encountered the following error(s): ['The response is no longer valid: either is too old or has been used previously.', 'unknown-error']",
-    ),
-}
-test_valid_response_from_api_data = {
-    "high_score": (high_score_return_value, True),
-    "low_score": (low_score_return_value, False),
-}
-
-
-def generate_fake_token() -> str:
-    return str(uuid.uuid4())
-
-
-@pytest.mark.parametrize(
-    "response_return_value,expected_result",
-    test_valid_response_from_api_data.values(),
-    ids=test_valid_response_from_api_data.keys(),
-)
-@patch("pcapi.connectors.api_recaptcha.RECAPTCHA_SECRET", "recaptcha-secret")
-@patch("pcapi.connectors.api_recaptcha.requests.post")
-def test_valid_response_from_api(request_post, response_return_value, expected_result):
-    # Given
-    token = generate_fake_token()
-    request_post.return_value = response_return_value
-
-    # When
-    api_response = validate_recaptcha_token(token, ORIGINAL_ACTION, 0.5)
-
-    # Then
-    request_post.assert_called_once_with(RECAPTCHA_API_URL, data={"secret": "recaptcha-secret", "response": token})
-    assert api_response == expected_result
-
-
-@patch("pcapi.connectors.api_recaptcha.requests.post")
-def test_with_empty_token(request_post):
-    # Given
-    request_post.return_value = errors_return_value
+from pcapi.connectors.api_recaptcha import get_token_validation_and_score
 
-    # When
-    api_response = validate_recaptcha_token(None, ORIGINAL_ACTION, 0.5)
 
-    # Then
-    request_post.assert_not_called()
-    assert api_response is False
+def _build_mocked_request_response(response: dict):
+    answer_with_weak_score = MagicMock(status_code=200, text="")
+    answer_with_weak_score.json = MagicMock(return_value=response)
```

Comment:
```suggestion
    answer_with_weak_score.json.return_value=response
```
un attribut d'un `MagickMock` est un `MagickMock`:
```
>>> answer_with_weak_score = MagicMock(status_code=200, text="")
>>> type(answer_with_weak_score.json)
<class 'unittest.mock.MagicMock'>
```

Reply:
Je recommande l'utilisation de `requests_mock` qui (d'après moi) permet d'avoir un mocking de `requests` plus lisible et sans le côté "free-for-all-yolo" des MagicMock (et, accessoirement, assez puissant). Je l'ai utilisé ici, par exemple https://github.com/pass-culture/pass-culture-api/blob/406ad62a1415edb9a8d900a44b273000b01fce82/tests/routes/pro/post_mediation_test.py#L40-L45.